File: src/kernel/limits.zig
//! Kernel capacity limits.
//!
//! Bounds for static kernel data structures. These limits exist because
//! the kernel uses fixed-size arrays during early boot before dynamic
//! allocation is available.

/// Maximum disjoint physical memory regions. Most systems have 1-2 regions
/// (main RAM, possibly high memory above 4GB). Systems with memory holes
/// or NUMA may have more. Cost: ~200 bytes per slot.
pub const MAX_MEMORY_ARENAS = 4;

/// Maximum reserved address ranges (kernel image, device tree, firmware).
pub const MAX_RESERVED_REGIONS = 8;

/// Maximum number of CPUs supported by static per-CPU structures.
pub const MAX_CPUS = 32;

File: kernel/src/limits.rs
//! Kernel capacity limits.
//!
//! Early boot uses static data structures. These bounds keep them fixed-size.

pub const MAX_MEMORY_ARENAS: usize = 4;
pub const MAX_RESERVED_REGIONS: usize = 8;
pub const MAX_TASKS: usize = 32;

pub const ENTRIES_PER_PAGE_TABLE: usize = 512;
pub const DEVICE_TREE_BLOB_MAX_SIZE: usize = 1 << 20;
pub const MINIMUM_PHYSMAP_SIZE: usize = 1 << 30;
